crates/edr_eth/src/spec.rs

```diff
@@ -27,6 +27,14 @@ pub trait ChainSpec {
     type SignedTransaction: ExecutableTransaction
         + revm_context_interface::Transaction
         + TransactionValidation;
+    /// The chain's way of building blocks
+    type BlockConstructor: BlockEnvConstructor<crate::block::Header, Self::BlockEnv>;
+}
+
+/// A trait for constructing a (partial) block header into an EVM block.
+pub trait BlockEnvConstructor<HeaderT, BlockT: Block> {
```

Comment:
I recommend making the `BlockT: Block` an associated type: `type BlockEnv: Block;` as it's constant per implementation of the trait (i.e. constant per ChainSpec).